python/cugraph/cugraph/tests/link_prediction/test_woverlap.py

```diff
@@ -14,158 +14,39 @@
 def setup_function():
     gc.collect()
 
 
-def cugraph_call(benchmark_callable, graph_file, pairs):
-    # Device data
-    cu_M = graph_file.get_edgelist()
-    weights_arr = cudf.Series(
-        np.ones(max(cu_M["src"].max(), cu_M["dst"].max()) + 1, dtype=np.float32)
-    )
-    weights = cudf.DataFrame()
-    weights["vertex"] = np.arange(len(weights_arr), dtype=np.int32)
-    weights["weight"] = weights_arr
-
-    G = graph_file.get_graph(create_using=cugraph.Graph(directed=True))
-
-    # cugraph Overlap Call
-    df = benchmark_callable(cugraph.overlap_w, G, weights, pairs)
-
-    df = df.sort_values(by=["first", "second"])
-    return df["overlap_coeff"].to_numpy()
-
-
-def intersection(a, b, M):
-    count = 0
-    a_idx = M.indptr[a]
-    b_idx = M.indptr[b]
-
-    while (a_idx < M.indptr[a + 1]) and (b_idx < M.indptr[b + 1]):
-        a_vertex = M.indices[a_idx]
-        b_vertex = M.indices[b_idx]
-
-        if a_vertex == b_vertex:
-            count += 1
-            a_idx += 1
-            b_idx += 1
-        elif a_vertex < b_vertex:
-            a_idx += 1
-        else:
-            b_idx += 1
-
-    return count
-
-
-def degree(a, M):
-    return M.indptr[a + 1] - M.indptr[a]
-
-
-def overlap(a, b, M):
-    b_sum = degree(b, M)
-    if b_sum == 0:
-        return float("NaN")
-
-    i = intersection(a, b, M)
-    a_sum = degree(a, M)
-    total = min(a_sum, b_sum)
-    return i / total
-
-
-def cpu_call(M, first, second):
-    result = []
-    for i in range(len(first)):
-        result.append(overlap(first[i], second[i], M))
-    return result
-
-
-@pytest.mark.sg
-@pytest.mark.parametrize("graph_file", UNDIRECTED_DATASETS)
-def test_woverlap(gpubenchmark, graph_file):
-    dataset_path = graph_file.get_path()
-    Mnx = utils.read_csv_for_nx(dataset_path)
-    N = max(max(Mnx["0"]), max(Mnx["1"])) + 1
-    M = scipy.sparse.csr_matrix((Mnx.weight, (Mnx["0"], Mnx["1"])), shape=(N, N))
-
-    G = graph_file.get_graph(ignore_weights=True)
-    pairs = (
-        G.get_two_hop_neighbors()
-        .sort_values(["first", "second"])
-        .reset_index(drop=True)
-    )
-
-    cu_coeff = cugraph_call(gpubenchmark, graph_file, pairs)
-    cpu_coeff = cpu_call(M, pairs["first"], pairs["second"])
-    assert len(cu_coeff) == len(cpu_coeff)
-    for i in range(len(cu_coeff)):
-        if np.isnan(cpu_coeff[i]):
-            assert np.isnan(cu_coeff[i])
-        elif np.isnan(cu_coeff[i]):
-            assert cpu_coeff[i] == cu_coeff[i]
-        else:
-            diff = abs(cpu_coeff[i] - cu_coeff[i])
-            assert diff < 1.0e-6
-
-
-@pytest.mark.sg
-@pytest.mark.parametrize("graph_file", UNDIRECTED_DATASETS)
-def test_woverlap_multi_column(graph_file):
+# =============================================================================
+# Pytest Fixtures
+# =============================================================================
+@pytest.fixture(scope="module", params=UNDIRECTED_DATASETS)
+def read_csv(request):
```

Comment:
This fixture doesn't have anything specific to these tests, so I'd move it to `conftest.py` so we're not copying it in so many test modules.  If that's too much work for this PR, I'd add a `FIXME`.